src/parser/highlight.ts

```diff
@@ -2,15 +2,16 @@ import hljs from 'highlight.js';
 
 export default function highlight(str: string, lang: string) {
     let content = str;
-    if (lang) {
-        if (hljs.getLanguage(lang)) {
-            try {
-                content = hljs.highlight(content, {
-                    language: lang,
-                    ignoreIllegals: true,
-                }).value;
-            } catch {}
-        }
+    if (!lang) {
+        lang = 'text';
+    }
+    if (hljs.getLanguage(lang)) {
+        try {
+            content = hljs.highlight(content, {
+                language: lang,
+                ignoreIllegals: true,
+            }).value;
+        } catch {}
     }
```

Comment:
If we do it like this there's still the edge case where `hljs.getLanguage` returns `undefined` for unknown languages, e.g.

``````markdown
```hehe
<h1>this will render as html</h1>
```
``````

so what we could also do instead would be:

```suggestion
    if (!hljs.getLanguage(lang)) lang = 'text';
    try {
        content = hljs.highlight(content, {
            language: lang,
            ignoreIllegals: true,
        }).value;
    } catch {}

```

Reply:
Yeah that's a good idea actually. I'll wrap this up in a nice commit now and test.

Does this mean both of the original nested `if` blocks were kind of useless? Would not the result been with just the try .. catch on its own?